app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -107,34 +119,67 @@ class TopicListAdapter(
-    val binding: WelcomeBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
+  private class WelcomeViewHolder(val binding: WelcomeBinding) : RecyclerView.ViewHolder(binding.root) {
     internal fun bind(userAppHistoryViewModel: UserAppHistoryViewModel) {
       binding.viewModel = userAppHistoryViewModel
     }
   }
 
-  private class PromotedStoryViewHolder(
-    val binding: PromotedStoryCardBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
-    internal fun bind(promotedStoryViewModel: PromotedStoryViewModel) {
-      binding.viewModel = promotedStoryViewModel
+  inner class PromotedStoryListViewHolder(val binding: PromotedStoryListBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(
+      activity: AppCompatActivity,
+      promotedStoryListViewModel: PromotedStoryListViewModel,
+      promotedStoryList: MutableList<PromotedStoryViewModel>
+    ) {
+      binding.viewModel = promotedStoryListViewModel
+      val promotedStoryAdapter = PromotedStoryListAdapter(promotedStoryList)
+      val horizontalLayoutManager = LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
+      binding.promotedStoryListRecyclerView.apply {
+        layoutManager = horizontalLayoutManager
+        adapter = promotedStoryAdapter
+      }
+
+      val snapHelper = PagerSnapHelper()
+      binding.promotedStoryListRecyclerView.layoutManager = horizontalLayoutManager
+      snapHelper.attachToRecyclerView(binding.promotedStoryListRecyclerView)
+
+      val padding48 = dipToPixels(48)
+      val padding16 = dipToPixels(16)
+      if (promotedStoryList.size > 1) {
+        binding.promotedStoryListRecyclerView.setPadding(padding16, 0, padding48, 0)
+      } else {
+        binding.promotedStoryListRecyclerView.setPadding(padding16, 0, padding16, 0)
+      }
     }
   }
 
-  private class AllTopicsViewHolder(
-    val binding: AllTopicsBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
+  private class AllTopicsViewHolder(binding: AllTopicsBinding) : RecyclerView.ViewHolder(binding.root) {
     internal fun bind() {
     }
   }
 
-  private class TopicListViewHolder(
-    val binding: TopicSummaryViewBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
-    internal fun bind(topicSummaryViewModel: TopicSummaryViewModel) {
+  inner class TopicListViewHolder(val binding: TopicSummaryViewBinding) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(topicSummaryViewModel: TopicSummaryViewModel, position: Int) {
       binding.viewModel = topicSummaryViewModel
+      val param = binding.topicContainer.layoutParams as GridLayoutManager.LayoutParams
+      val margin32 = dipToPixels(32)
+      val margin8 = dipToPixels(8)
+      if (position % 2 == 0) {
+        param.setMargins(margin8, margin8, margin32, margin8)
+
+      } else {
+        param.setMargins(margin32, margin8, margin8, margin8)
+      }
+      binding.topicContainer.layoutParams = param
     }
   }
+
+  fun dipToPixels(dipValue: Int): Int {
```

Comment:
Can be private.

Also, prefer using: https://developer.android.com/reference/android/content/res/Resources#getDimensionPixelSize(int) with an XML-stored dimension. That might be a bit cleaner, and lets you potentially share dimensions between Kotlin & XML if it makes sense.

Reply:
Removed this method and changed code to 

`(activity as Context).resources.getDimensionPixelSize(R.dimen.padding_48)`
 and similarly for others in this file.